### PR TITLE
Starts postgresql SSL procedure

### PR DESCRIPTION
**Pull request by stevsmit**

### CODE REVIEW

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+. After you have generated the CAs and SSL/TLS certificates and keys for your CloudSQL database and ensured that they are in `.pem` format, test the SSL connection to your CloudSQL server:
+
+.. Initiate a connection to your CloudSQL server by entering the following command:
++
+[source,terminal]
+----
+$ psql "sslmode=verify-ca sslrootcert=<ssl_server_certificate_authority>.pem sslcert=<ssl_client_certificate>.pem sslkey=<ssl_client_key>.pem hostaddr=<34.29.130.58> port=<5432> user=<cloudsql_username> dbname=<cloudsql_database_name>"
+----
+
+.. Create a new database and verify that the connection is using SSL by running the following command:
```
Comment (SeanZhao-redhat) [inline modules/configuring-cert-based-auth-quay-cloudsql.adoc:26]: this part is optional, I think we needn't so detail, as I think this should be backend knowledge and common sense to use quay

Comment (stevsmit) [reply modules/configuring-cert-based-auth-quay-cloudsql.adoc:26]: @SeanZhao-redhat Which part? Lines 16-76? or just line 26?

Thanks!

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+* You have `base64 decoded` the original config bundle into a `config.yaml` file. For more information, see link:https://docs.redhat.com/en/documentation/red_hat_quay/{producty}/html-single/deploying_the_red_hat_quay_operator_on_openshift_container_platform/index#operator-config-cli-download[Downloading the existing configuration].
+* You are using an externally managed PostgreSQL or CloudSQL database. For more information, see link:https://docs.redhat.com/en/documentation/red_hat_quay/{producty}/html-single/deploying_the_red_hat_quay_operator_on_openshift_container_platform/index#operator-unmanaged-postgres[Using and existing PostgreSQL database] with the `DB_URI` variable set.
+* The `postgres` component of your `QuayRegistry` CRD is set to `managed: false`, and your CloudSQL database is set with the `DB_URI` configuration variable. The following procedure uses `postgresql://<cloudsql_username>:<dbpassword>@34.29.130.58:5432/quay14`.
+* For CloudSQL, *Manage SSL mode* must be set to *Require trusted client certificates*. This can be set on the *Connections* page of your CloudSQL instance.
```
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/configuring-cert-based-auth-quay-cloudsql.adoc:13]: Removed line 13.

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc (hunk 1 above)
Comment (SeanZhao-redhat) [reply modules/configuring-cert-based-auth-quay-cloudsql.adoc:26]: Lines 16-76 are optional, I review again, line 16~25 looks good, I suggest we keep it, remove line 26~76

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+* You have `base64 decoded` the original config bundle into a `config.yaml` file. For more information, see link:https://docs.redhat.com/en/documentation/red_hat_quay/{producty}/html-single/deploying_the_red_hat_quay_operator_on_openshift_container_platform/index#operator-config-cli-download[Downloading the existing configuration].
+* You are using an externally managed PostgreSQL or CloudSQL database. For more information, see link:https://docs.redhat.com/en/documentation/red_hat_quay/{producty}/html-single/deploying_the_red_hat_quay_operator_on_openshift_container_platform/index#operator-unmanaged-postgres[Using and existing PostgreSQL database] with the `DB_URI` variable set.
+* Your externally managed PostgreSQL or CloudSQL database is configured for SSL/TLS.
+* The `postgres` component of your `QuayRegistry` CRD is set to `managed: false`, and your CloudSQL database is set with the `DB_URI` configuration variable. The following procedure uses `postgresql://<cloudsql_username>:<dbpassword>@34.29.130.58:5432/quay14`.
```
Comment (SeanZhao-redhat) [inline modules/configuring-cert-based-auth-quay-cloudsql.adoc:13]: 34.29.130.58 is myself test instance ip, quay14 is my db, should change to more generic, like: 
`postgresql://<cloudsql_username>:<dbpassword>@<dbhost>:<port>/<database>`.

pls do a search, this also happens in next content

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+DB_CONNECTION_ARGS:
+    autorollback: true
+    sslmode: verify-ca <1>
+    sslrootcert: /.postgresql/<root.crt> <2>
+    sslcert: /.postgresql/<cloudsql.crt> <3>
+    sslkey: /.postgresql/<cloudsql.key> <4>
+    threadlocals: true <5>
+DB_URI: postgresql://<dbusername>:<dbpassword>@34.29.130.58:5432/quay14?sslmode=verify-ca&sslcert=/.postgresql/<cloudsql.crt>&sslkey=/.postgresql/<cloudsql.key>&sslrootcert=/.postgresql/<root.crt> <6>
+----
+<1> Ensures that the database only tries an SSL/TLS connection and verifies that the server certificate is issued by a trusted CA.
```
Comment (SeanZhao-redhat) [inline modules/configuring-cert-based-auth-quay-cloudsql.adoc:121]: no, both trusted CA and self-signed CA can work with `verify-ca`, if the hostname matches a SAN entry in your server certificate, then can use `verify-full`.

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+<2> The path of the `root.crt` file, which contains the root certificate used to verify the SSL/TLS connection with your CloudSQL database. This file is mounted in the `Quay` pod from the Kubernetes secret.
+<3> The path of the `cloudsql.crt` file, which contains the client certificate used to authenticate the connection to your CloudSQL database. This file is mounted in the `Quay` pod from the Kubernetes secret.
+<4> The path of the `cloudsql.key` file, which contains the private key associated with the client certificate. This file is mounted in the `Quay` pod from the Kubernetes secret.
+<5> Enables auto-rollback for connections.
+<6> The URI that accesses your CloudSQL database appended with your `root.crt`, `cloudsql.crt`, and `cloudsql.key` files. If using `verify-ca` as your SSL mode, you must also include the database username and password. The SSL/TLS information included in `DB_URI` must match the information provided in `DB_CONNECTION_ARGS`.
```
Comment (SeanZhao-redhat) [inline modules/configuring-cert-based-auth-quay-cloudsql.adoc:126]: "If using `verify-ca` as your SSL mode, you must also include the database username and password" -> no, this is not related with `verify-ca`, this is GCP CloudSQL specific: it requires database username and password in `DB_URI`

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+.. To verify the certificate against a trusted CA *and* match a SAN name in your certificate, use the `verify-full` SSL mode. Edit your `config.yaml` file to include the following information:
```
Comment (SeanZhao-redhat) [inline modules/configuring-cert-based-auth-quay-cloudsql.adoc:128]: We needn't separate `verify-ca` and `verify-full` , it's certs itself property, the certificates are user provided, they should clear which to parameter to use, they can config anyone based on there certs

Comment (SeanZhao-redhat) [reply modules/configuring-cert-based-auth-quay-cloudsql.adoc:128]: A simple way maybe, not explain `verify-ca` and `verify-full` in the whole context

Comment (SeanZhao-redhat) [reply modules/configuring-cert-based-auth-quay-cloudsql.adoc:128]: @stevsmit this part (line 128 -> 194), are you want to express? As Michaela mentioned  "`2) Connectivity ONLY through Certificate Authentication`"
```
DB_CONNECTION_ARGS:
    autorollback: true
    sslmode: verify-full
    sslrootcert: conf/stack/database.pem
    threadlocals: true
DB_URI: postgresql://<dbusername>:<dbpassword>@quay-sean-postgres16.cmqwuswughvh.us-east-2.rds.amazonaws.com:5432/quay?sslmode=verify-full&sslrootcert=/quay-registry/conf/stack/database.pem
```
if yes, then pls category as she mentioned, something like "Connectivity and Authentication" and "Connectivity Only" , if no, this part is redundant.

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc (hunk 3 above)
Comment (stevsmit) [reply modules/configuring-cert-based-auth-quay-cloudsql.adoc:13]: Fixed both instances.

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc (hunk 5 above)
Comment (stevsmit) [reply modules/configuring-cert-based-auth-quay-cloudsql.adoc:126]: Fixed to 

<6> The URI that accesses your CloudSQL database appended with your `root.crt`, `cloudsql.crt`, and `cloudsql.key` files. The SSL/TLS information included in `DB_URI` must match the information provided in `DB_CONNECTION_ARGS`. If you are using CloudSQL, you must include your database username and password in this variable.

File: modules/configuring-cert-based-auth-quay-cloudsql.adoc [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+.Procedure
+
+. After you have generated the CAs and SSL/TLS certificates and keys for your CloudSQL database and ensured that they are in `.pem` format, test the SSL connection to your CloudSQL server:
+
+.. Initiate a connection to your CloudSQL server by entering the following command:
++
+[source,terminal]
+----
+$ psql "sslmode=verify-ca sslrootcert=<ssl_server_certificate_authority>.pem sslcert=<ssl_client_certificate>.pem sslkey=<ssl_client_key>.pem hostaddr=<34.29.130.58> port=<5432> user=<cloudsql_username> dbname=<cloudsql_database_name>"
```
Comment (SeanZhao-redhat) [inline modules/configuring-cert-based-auth-quay-cloudsql.adoc:23]: `hostaddr=<34.29.130.58>` better to be generic:  `hostaddr=<database_host>`

**Comment by SeanZhao-redhat** [comment]
lgtm
